tap_zendesk/http.py

```diff
@@ -208,7 +210,80 @@ def get_offset_based(url, access_token, request_timeout, page_size, **kwargs):
         yield response_json
         next_url = response_json.get('next_page')
 
-def get_incremental_export(url, access_token, request_timeout, start_time):
+@backoff.on_exception(backoff.expo,
+                    (ConnectionError, ConnectionResetError, Timeout, ChunkedEncodingError, ProtocolError),#As ConnectionError error and timeout error does not have attribute status_code,
+                    max_tries=5, # here we added another backoff expression.
+                    factor=2)
+async def call_api_async(session, url, request_timeout, params, headers):
+    """
+    Perform an asynchronous GET request
+    """
+    while True:
+        async with session.get(url, params=params, headers=headers, timeout=request_timeout) as response:
+            try:
+                response_json = await response.json()
+            except Exception: # pylint: disable=broad-except
+                response_json = {}
```

Comment:
The except Exception block is too broad. It would be better to catch specific exceptions. Also in case of exception we should log the warning.

Reply:
I have created a separate function `raise_for_error_for_async` for the response validation.